app/v1/support/DesResponseMappingSupport.scala

```diff
@@ -20,10 +20,26 @@ import utils.Logging
 import v1.controllers.EndpointLogContext
 import v1.models.errors._
 import v1.models.outcomes.ResponseWrapper
+import v1.models.response.retrieveBusinessDetails.{RetrieveBusinessDetailsResponse, RetrieveBusinessDetailsResponseList}
 
 trait DesResponseMappingSupport {
   self: Logging =>
 
+  final def filterId(
+    responseWrapper: ResponseWrapper[RetrieveBusinessDetailsResponseList],
+    businessId: String
+  ): Either[ErrorWrapper, ResponseWrapper[RetrieveBusinessDetailsResponse]] = {
+    val filteredBusinesses = responseWrapper.responseData.retrieveBusinessDetailsResponse.filter {
+      BusinessDetails => businessId == BusinessDetails.businessId
+    }
+
+    if (filteredBusinesses.nonEmpty) {
+      Right(ResponseWrapper(responseWrapper.correlationId, filteredBusinesses.head))
```

Comment:
I don't like this... what will happen if there's more than one returned for some reason? Have a look at what I've done in the [crystallisation obligations endpoint](https://github.com/hmrc/obligations-api/blob/master/app/v1/support/DesResponseMappingSupport.scala#L77-L86) - something like
```lang-scala
filteredBusinesses match {
  case business :: Nil => Right(ResponseWrapper(responseWrapper.correlationId, business))
  case Nil => Left(ErrorWrapper(Some(responseWrapper.correlationId), NoBusinessFoundError))
  case _ :: _ => ??? // whatever happens where there is more than one returned
                     // check with Francis if this could ever happen
                     // if it can happen then I assume we will use DownstreamError
}
```
It avoids the ugly `.head` and accounts for there being more than one returned